### ISSUE TITLE
Incomplete file copy from cloud to local file system

### ISSUE THREAD

**Issue by dan-cristian** [TurboGit/hubicfuse#87]
when I copy a file from a hubicfuse mount to local fs I get only partial content copied, size is randomly variable, always smaller that real size on the cloud. Any advice please?

on the cloud I have: -rw-rw-rw- 1 haiot haiot 27807454 Nov  1 23:38 webmin_1.770_all.deb

when I copy, with cp, with debug enabled, hubicfuse stops at offset 1269060, so local file looks like:
-rw-r--r--  1 haiot haiot 1269060 Nov  6 01:56 webmin_1.770_all.deb

   open[139691730930400] flags: 0x8000 /backup/webmin_1.770_all.deb
   unique: 28, success, outsize: 32
unique: 29, opcode: GETATTR (3), nodeid: 3, insize: 56, pid: 29275
getattr /backup/webmin_1.770_all.deb
   unique: 29, success, outsize: 120
unique: 30, opcode: READ (15), nodeid: 3, insize: 80, pid: 29275
read[139691730930400] 131072 bytes from 0 flags: 0x8000
   read[139691730930400] 131072 bytes from 0
   unique: 30, success, outsize: 131088
unique: 31, opcode: READ (15), nodeid: 3, insize: 80, pid: 29275
read[139691730930400] 131072 bytes from 131072 flags: 0x8000
   read[139691730930400] 131072 bytes from 131072
   unique: 31, success, outsize: 131088
unique: 32, opcode: READ (15), nodeid: 3, insize: 80, pid: 29275
read[139691730930400] 131072 bytes from 262144 flags: 0x8000
   read[139691730930400] 131072 bytes from 262144
   unique: 32, success, outsize: 131088
unique: 33, opcode: READ (15), nodeid: 3, insize: 80, pid: 29275
read[139691730930400] 131072 bytes from 393216 flags: 0x8000
   read[139691730930400] 131072 bytes from 393216
   unique: 33, success, outsize: 131088
unique: 34, opcode: READ (15), nodeid: 3, insize: 80, pid: 29275
read[139691730930400] 131072 bytes from 524288 flags: 0x8000
   read[139691730930400] 131072 bytes from 524288
   unique: 34, success, outsize: 131088
unique: 35, opcode: READ (15), nodeid: 3, insize: 80, pid: 29275
read[139691730930400] 131072 bytes from 655360 flags: 0x8000
   read[139691730930400] 131072 bytes from 655360
   unique: 35, success, outsize: 131088
unique: 36, opcode: READ (15), nodeid: 3, insize: 80, pid: 29275
read[139691730930400] 131072 bytes from 786432 flags: 0x8000
   read[139691730930400] 131072 bytes from 786432
   unique: 36, success, outsize: 131088
unique: 37, opcode: READ (15), nodeid: 3, insize: 80, pid: 29275
read[139691730930400] 131072 bytes from 917504 flags: 0x8000
   read[139691730930400] 131072 bytes from 917504
   unique: 37, success, outsize: 131088
unique: 38, opcode: READ (15), nodeid: 3, insize: 80, pid: 29275
read[139691730930400] 131072 bytes from 1048576 flags: 0x8000
   read[139691730930400] 131072 bytes from 1048576
   unique: 38, success, outsize: 131088
unique: 39, opcode: READ (15), nodeid: 3, insize: 80, pid: 29275
read[139691730930400] 131072 bytes from 1179648 flags: 0x8000
   read[139691730930400] 89412 bytes from 1179648
   unique: 39, success, outsize: 89428
unique: 40, opcode: READ (15), nodeid: 3, insize: 80, pid: 29275
read[139691730930400] 131072 bytes from 1269060 flags: 0x8000
   read[139691730930400] 0 bytes from 1269060
   unique: 40, success, outsize: 16
unique: 41, opcode: FLUSH (25), nodeid: 3, insize: 64, pid: 29275
flush[139691730930400]
   unique: 41, success, outsize: 16
unique: 42, opcode: RELEASE (18), nodeid: 3, insize: 64, pid: 0
release[139691730930400] flags: 0x8000
   unique: 42, success, outsize: 16

**Comment by TurboGit** [comment]
We would need more information as many people are using hubicfuse without having truncated files :) Is that with your patch (discussed in another thread)? Which OS? Which version of the OS? How do you mount?

**Comment by dan-cristian** [comment]
I have tried this on a different machine (Debian jessie/Intel x64) and it works well so far, including the patched version with RETRY_NUMBER=1 and 1 second sleep on retry.

I think the issue I had on my first machine seems to be because of the OS, it is Hyper-V (debian, jessie) running on WIndows 8.1 Enterprise, NAT network, Intel x64, i7. 

haiot@debian:~$ uname -a
Linux debian 3.16.0-4-amd64 #1 SMP Debian 3.16.7-ckt11-1+deb8u5 (2015-10-09) x86_64 GNU/Linux

hubic mounted as:
hubicfuse /mnt/hubic -o noauto_cache,sync_read,allow_other -d

It is strange, as I have tested again now and works fine, seems a random behaviour. Will keep investigate.
